cmake/CompilerWarnings.cmake

```diff
@@ -80,9 +73,15 @@ function(set_target_warnings target)
         )
     endif()
 
-    if(SFML_COMPILER_CLANG)
+    if(SFML_COMPILER_CLANG OR SFML_COMPILER_CLANG_CL)
         target_compile_options(${target} PRIVATE
             -Wno-unknown-warning-option # do not warn on GCC-specific warning diagnostic pragmas
         )
     endif()
+
+    # Disable certain deprecation warnings
+    if(SFML_OS_WINDOWS)
```

Comment:
Should those really be enabled for all Windows compilers?

Reply:
Yes, because we need to disable those deprecation warnings for MSVC, clang-cl, and Clang.